popmon/analysis/profiling/profiles.py

```diff
@@ -16,86 +16,189 @@
+    dim=1,
+)
+def profile_underflow(hist):
+    if hasattr(hist, "underflow"):
+        return hist.underflow.entries
+    return 0
+
+
+@Profiles.register(
+    key="phik",
+    description="phi-k correlation between the two variables of the histogram",
+    dim=2,
+)
+def profile_phik(hist):
+    from phik import phik
+
+    # calculate phik correlation
+    try:
+        grid = get_2dgrid(hist)
+    except Exception:
+        raise
+
+    try:
+        phi_k = phik.phik_from_hist2d(observed=grid)
+    except ValueError:
+        # self.logger.debug(
+        #     f"Not enough values in the 2d `{name}` time-split histogram to apply the phik test."
+        # )
+        phi_k = np.nan
+    return phi_k
+
+
+@Profiles.register(
+    key="count", description="Number of entries (non-NaN and NaN)", dim=None
+)
+def profile_count(hist):
+    return int(sum_entries(hist))
+
+
+@Profiles.register(
+    key="filled",
+    description="Number of non-missing entries (non-NaN)",
+    dim=1,
+    htype="all",
+)
+def profile_filled(_, bin_counts):
+    return bin_counts.sum()
+
+
+@Profiles.register(
+    key="distinct", description="Number of distinct entries", dim=1, htype="all"
+)
+def profile_distinct(bin_labels, bin_counts):
+    return len(np.unique(bin_labels[bin_counts > 0]))
+
+
+def fraction_of_true(bin_labels, bin_entries):
+    """Compute fraction of 'true' labels
+
+    :param bin_labels: Array containing numbers whose mean is desired. If `a` is not an
+        array, a conversion is attempted.
+    :param bin_entries: Array containing weights for the elements of `a`. If `weights` is not an
+        array, a conversion is attempted.
+    :return: fraction of 'true' labels
+    """
+    bin_labels = np.array(bin_labels)
+    bin_entries = np.array(bin_entries)
+    assert len(bin_labels) == len(bin_entries)
+
+    def replace(bl):
+        if bl in {"True", "true"}:
+            return True
+        elif bl in {"False", "false"}:
+            return False
+        return np.nan
+
+    # basic checks: dealing with boolean labels
+    # also accept strings of 'True' and 'False'
+    if len(bin_labels) == 0 or len(bin_labels) > 4 or np.sum(bin_entries) == 0:
+        return np.nan
+    if not np.all([isinstance(bl, (bool, np.bool_)) for bl in bin_labels]):
+        if not np.all(
+            [isinstance(bl, (str, np.str_, np.string_)) for bl in bin_labels]
+        ):
+            return np.nan
+        # all strings from hereon
+        n_true = (bin_labels == "True").sum() + (bin_labels == "true").sum()
+        n_false = (bin_labels == "False").sum() + (bin_labels == "false").sum()
+        n_nan = (
+            (bin_labels == "NaN").sum()
+            + (bin_labels == "nan").sum()
+            + (bin_labels == "None").sum()
+            + (bin_labels == "none").sum()
+            + (bin_labels == "Null").sum()
+            + (bin_labels == "null").sum()
```

Comment:
I think using np.unique(bin_labels, return_counts=True) would be way faster than six comparison operations.
counts = dict(zip(*np.unique(bin_labels, return_counts=True)))
n_nan = counts["NaN"] + counts["nan"] + counts["None"] + counts["none"] + counts["Null"] + counts["null"]

Reply:
Could indeed be cleaned. The value counts should already be present in the histogram anyway.